bin/test-lxd-devlxd-vm

```diff
@@ -107,7 +108,10 @@ echo "==> Checking devlxd is not working"
 
 ! lxc exec v1 -- curl -s --unix-socket /dev/lxd/sock http://custom.socket/1.0 || false
 
-lxc restart -f v1
+# Let some time pass before restarting the VM again, otherwise it won't know that it needs to restart.
+sleep 5
```

Comment:
Why is the sleep 5 needed here before the restart, the VM should be up and running by this point because you used `waitVMAgent` above.

Reply:
That should be the case, but unfortunately it isn't. Adding a short sleep, let's it restart.

Reply:
Can you try execing a `sync` command inside the VM before doing a forceful restart to see if that fixes this, without needing the graceful restart and sleeping.

Reply:
Adding `sync` fixes all our problems. I'll update the PR.